internal/extclient/terraform_cloud.go

```diff
@@ -0,0 +1,52 @@
+package extclient
```

Comment:
Apart from the error messages, this `TerraformCloudAPIClient`  seems very generic. It's essentially a wrapper around a HTTP client that adds a bearer token. I think we remove the Terraform specific messages, rename the package to `http` and rename the struct to `AuthedClient`